distributed/comm/tests/test_ucx.py

```diff
@@ -0,0 +1,313 @@
+import asyncio
+import itertools
+
+import pytest
+
+ucp = pytest.importorskip("ucp")
+
+from distributed import Client
+from distributed.comm import ucx, listen, connect
+from distributed.comm.registry import backends, get_backend
+from distributed.comm import ucx, parse_address
+from distributed.protocol import to_serialize
+from distributed.deploy.local import LocalCluster
+from distributed.utils_test import gen_test, loop, inc  # noqa: 401
+
+from .test_comms import check_deserialize
+
+
+HOST = ucp.get_address()
+
+# Currently having some issues with re-using ports.
+# Tests just hang. Still debugging.
+port_counter = itertools.count(13337)
+
+
+def test_parse_address():
+    result = ucx._parse_address("ucx://10.33.225.160")
+    assert result == ("ucx", "10.33.225.160")
+
+
+def test_parse_host_port():
+    assert ucx._parse_host_port("10.33.225.160:13337") == ("10.33.225.160", 13337)
+    assert ucx._parse_host_port("10.33.225.160:13338") == ("10.33.225.160", 13338)
+
+
+def test_registered():
+    assert "ucx" in backends
+    backend = get_backend("ucx")
+    assert isinstance(backend, ucx.UCXBackend)
+
+
+async def get_comm_pair(listen_addr, listen_args=None, connect_args=None, **kwargs):
+    q = asyncio.queues.Queue()
+
+    async def handle_comm(comm):
+        await q.put(comm)
+
+    # Workaround for hanging test in
+    # pytest distributed/comm/tests/test_ucx.py::test_comm_objs -vs --count=2
+    # on the second time through.
+    ucp._libs.ucp_py.reader_added = 0
```

Comment:
@Akshay-Venkatesh do you recall if this was resolved? Is it the same as https://github.com/Akshay-Venkatesh/ucx-py/issues/69, or different?

Reply:
@TomAugspurger There was a fix missing for the test but the test is passing.

Reply:
@TomAugspurger I tested this yesterday and the issue hasn't been resolved. This still has to be fixed.